times_reader/transforms.py

```diff
@@ -353,50 +351,52 @@ def remove_comment_rows(table: datatypes.EmbeddedXlTable) -> datatypes.EmbeddedX
         return table
 
     df = table.dataframe.copy()
-    comment_rows = list(
-        locate(
-            df.iloc[:, 0],
-            lambda cell: isinstance(cell, str)
-            and (cell.startswith("*") or cell.startswith("\\I:")),
-        )
-    )
-    df.drop(index=comment_rows, inplace=True)
-    df.reset_index(drop=True, inplace=True)
+    colname = df.columns[0]
+
+    tag = table.tag.split(":")[0]
+
+    if tag in config.row_comment_chars:
+        chars_by_colname = config.row_comment_chars[tag]
+    else:
+        return table
 
-    # TODO: the deletion of this third column is a bug. Removing it causes the
-    # program to fail parse all rows. We need to fix the parsing so it can read
-    # all rows and remove this code block.
-    if df.shape[1] > 1:
-        comment_rows = list(
-            locate(
-                df.iloc[:, 1],
-                lambda cell: isinstance(cell, str) and cell.startswith("*"),
+    for colname in df.columns:
+        if colname in chars_by_colname.keys():
+            comment_rows = list(
+                locate(
+                    df[colname],
+                    lambda cell: isinstance(cell, str)
+                    and (cell.startswith(tuple(chars_by_colname[colname]))),
+                )
             )
-        )
-        df.drop(index=comment_rows, inplace=True)
-        df.reset_index(drop=True, inplace=True)
+            df.drop(index=comment_rows, inplace=True)
+            df.reset_index(drop=True, inplace=True)
```

Comment:
And if we do want to delete rows with comment chars in any column, it might be more efficient to build up the `comment_rows` list in the loop and do the `drop` just once at the end of the loop.

Reply:
Done!